alternator/server.cc

```diff
@@ -278,6 +278,10 @@ future<executor::request_return_type> server::handle_api_request(std::unique_ptr
             _executor._stats.unsupported_operations++;
             throw api_error::unknown_operation(format("Unsupported operation {}", op));
         }
+        if (_pending_requests.get_count() >= _max_concurrent_requests) {
+            return make_ready_future<executor::request_return_type>(
+                    api_error::request_limit_exceeded(format("too many in-flight requests: {}", _pending_requests.get_count())));
```

Comment:
I assume these counters and limits are per-shard, but the variable names and the error message don't make this clear. You're sending a user a message which looks like "too many in-flight requests: 10,000" while since there are 10 shards, the actual limit for the entire system is 100,000, which may be confusing.
Also, I think it would be nice for the error message to explain where this limit is configured, in case the user thinks this is a mistake and wants to fix the configuration.

Reply:
I just added it to CQL per Botond's request, and am in the middle of changing it for alternator too :) I'll push v2 in a minute